### PR TITLE
Only fire negotiationneeded when chain is/becomes empty.

### PR DESCRIPTION
**Pull request by jan-ivar**
Fixes https://github.com/w3c/webrtc-pc/issues/2508.

***
<a href="https://pr-preview.s3.amazonaws.com/jan-ivar/webrtc-pc/pull/2509.html" title="Last updated on Apr 16, 2020, 6:42 PM UTC (fbac0c4)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/webrtc-pc/2509/544a656...jan-ivar:fbac0c4.html" title="Last updated on Apr 16, 2020, 6:42 PM UTC (fbac0c4)">Diff</a>

### CODE REVIEW

**Comment by jan-ivar** [comment]
This PR is based on implementation being tested and ready to land in Firefox in [bug 1628139](https://bugzil.la/1628139) with WPT tests.

This is blocking and is needed to green up the [perfect negotiation WPT tests](https://bugzil.la/1621399).

File: webrtc.html [hunk 1]
```diff
@@ -3850,56 +3864,66 @@ <h4>Updating the Negotiation-Needed flag</h4>
         implementation that affect negotiation. If such changes occur, the user
         agent MUST [= update the negotiation-needed flag =].</p>
         <p data-tests="">To <dfn>update the negotiation-needed flag</dfn> for
-        <var>connection</var>, queue a task to [= chain =] the following steps
-        to <var>connection</var>'s [= operations chain =]:</p>
+        <var>connection</var>, run following steps:</p>
```
Comment (alvestrand) [inline webrtc.html:3867]: Nit: "run the following steps"

File: webrtc.html [hunk 2]
```diff
@@ -1078,7 +1083,16 @@ <h4>Chain an asynchronous operation</h4>
                   <li>
                     <p>If <a>[[\Operations]]</a> is non-empty, execute the
                     operation represented by the first element of
-                    <a>[[\Operations]]</a>.</p>
+                    <a>[[\Operations]]</a>, and abort these steps.</p>
+                  </li>
+                  <li>
```
Comment (alvestrand) [inline webrtc.html:1088]: Missing read of the flag and "abort these steps" here?

**Comment by jan-ivar** [comment]
Thanks @alvestrand for catching the missing check! Fixed.

**Pull request merged**